ExtractUserDataActivity/handler.ts

```diff
@@ -276,22 +252,64 @@ export const getAllMessagesStatuses = (
 ): TaskEither<ActivityResultQueryFailure, ReadonlyArray<MessageStatus>> =>
   array.sequence(taskEither)(
     messages.map(({ id: messageId }) =>
-      fromQueryEither(
-        () => messageStatusModel.findOneByMessageId(messageId),
-        "messageStatusModel.findOneByMessageId"
-      ).foldTaskEither<ActivityResultQueryFailure, MessageStatus>(
-        failure => fromEither(left(failure)),
-        maybeContent =>
-          fromEither(
-            maybeContent.foldL(
-              () => right({ messageId } as MessageStatus),
-              content => right(content)
+      messageStatusModel
+        .findLastVersionByModelId(messageId)
+        .foldTaskEither<ActivityResultQueryFailure, MessageStatus>(
+          failure =>
+            fromEither(
+              left(
+                ActivityResultQueryFailure.encode({
+                  kind: "QUERY_FAILURE",
+                  reason: `messageStatusModel|${failure.kind}` // FIXME: better mapping from CosmosErrors to ActivityResultQueryFailure?
+                })
+              )
+            ),
+          maybeContent =>
+            fromEither(
+              maybeContent.foldL(
+                () => right({ messageId } as MessageStatus),
+                content => right(content)
+              )
             )
-          )
-      )
+        )
     )
   );
 
+export const findNotificationsForMessage = (
```

Comment:
can we use this one instead https://github.com/pagopa/io-functions-commons/blob/master/src/models/notification.ts#L168 ?

Reply:
Nope, this methods returns only one result [https://github.com/pagopa/io-functions-commons/blob/5cf3fc260678491ef94c9c4f570d559eccd0d56c/src/models/notification.ts#L171](https://github.com/pagopa/io-functions-commons/blob/5cf3fc260678491ef94c9c4f570d559eccd0d56c/src/models/notification.ts#L171). Here we have to retrieve a set of notifications.

Reply:
the relationship between messages and notification is 1-to-1 so this query will always return just one result

Reply:
Ah ok, i was mislead by method's naming -> `findNotificationsForMessage` vs `findNotificationForMessage`